### ISSUE TITLE
evil-paste-pop, highlight remains and can't be cleared easily

### ISSUE THREAD

**Issue by duianto** [edkolev/evil-goggles#30]
### Description
With an active region, and having copied two pieces of text `y`.
Pasting `p` and replacing the paste with: `M-y` (`evil-paste-pop`)

Results in the highlight remaining visible, and it can't be cleared easily.

One way is to reopen the buffer, or for a temporary buffer,
copy it's contents and paste it in a new temporary buffer.

### Reproduction steps
With `evil` and `evil-goggles` installed.
`init.el` contains:
```
  (require 'evil)
  (evil-mode 1)

  (evil-goggles-mode)

  (setq evil-goggles-pulse nil)
```

- Create a new buffer: `C-x b` `test` `RET`
- Type:
```
a
b
c
```
- Copy the "c" line: `yy`
- Move up to the "b" line: `k`
- Copy the "b" line: `yy`
- Move up to the "a" line: `k`
- Select the "a": `v`
- Paste: `p`
the buffer contains:
```

b

b
c
```
- replace the paste with the previously copied line: `M-y`
it calls: `evil-paste-pop`

### Observed
The buffer contains:
```

c

b
c
```
the pasted "c" line is highlighted.

![emacs_2021-03-22_21-17-17](https://user-images.githubusercontent.com/13420573/112053537-c1363100-8b54-11eb-9689-6abfb115e8d7.png)

### Expected
The highlight shouldn't remain.

And/or it should at least be cleared when another operation is performed 
that shows and hides a `evil-goggles` highlight.

### Notes

That the highlight is evil-goggles related can be as follows:
- Place the cursor on the highlight and call: `M-x` `describe-char` `RET`
- Switch to the `*Help*` buffer: `C-w j` (or `C-x o`)
- Scroll to the bottom: `G` (or `M->`)

The overlay says:
```
There is an overlay here:
 From 2 to 4
  face                 evil-goggles-default-face
  insert-behind-hooks  (evil-goggles--overlay-insert-behind-hook)
  priority             9999
  window               #<window 3 on test>
```

Deleting the highlighted line: `dd`
and describing the char again: `M-x` `describe-char` `RET`
shows that it's still there:
```
There is an overlay here:
 From 2 to 2
  face                 evil-goggles-default-face
  insert-behind-hooks  (evil-goggles--overlay-insert-behind-hook)
  priority             9999
  window               #<window 3 on test>
```

Typing at that location shows the highlight on the typed characters.
For example when typing:
```
abc
def
```
The buffer now looks like this:
```

abc
def
b
c
```
the "abc" line and the letters "def" on the line below are highlighted.

![emacs_2021-03-22_21-19-10](https://user-images.githubusercontent.com/13420573/112053514-baa7b980-8b54-11eb-834f-3ae9ecd3faab.png)

The only way I've found to remove the highlight, 
is to reopended the buffer.

In this case the buffer isn't a saved file.
One way to remove the highlight is to copy the whole buffer
and paste it in a new temporary buffer.

### System Info
evil-goggles-20200101.1935
Evil version 1.14.0
GNU Emacs 27.1 (build 1, x86_64-w64-mingw32) of 2020-08-21
Windows 10 2004

**Comment by davidhaley** [comment]
I'm experiencing the same issue. I'm not quite sure how to reset the faces after performing `evil-paste-pop`.

**Comment by davidhaley** [comment]
> I'm experiencing the same issue. I'm not quite sure how to reset the faces after performing `evil-paste-pop`.

Just an update. I found a workaround for now. Although not ideal, `revert-buffer` resets the highlighted background (`SPC b r` in Doom Emacs).

**Comment by edkolev** [comment]
I can't reproduce - issue might be caused by Doom. Could you help me reproduce? Some ideas:
- do you get an error if you have `M-x toggle-debug-on-error` on
- do you have any advices for function `evil-paste-after` - check with `M-x describe-function RET evil-paste-after`

**Comment by edkolev** [comment]
I managed to reproduce it and fix it - thank you for the detailed bug report!

The fix is in the master branch, let me know in case it's still reproducible on your end.

**Comment by duianto** [comment]
Confirmed, it's fixed in: evil-goggles-20231021.738
There's no lingering highlight anymore. Thanks.

![emacs_QCGDrcv1Rv](https://github.com/edkolev/evil-goggles/assets/13420573/c6291d81-ae5e-4d38-8762-5f9d51280be1)

![emacs_hu7S9h7Alr](https://github.com/edkolev/evil-goggles/assets/13420573/ff244424-7bc9-4a50-82d9-03de74a4f5b5)
